psi4/share/psi4/fsapt/fsapt.py

```diff
@@ -328,40 +390,110 @@ def printFrag(geom, Z, Q, fragkeys, frags, nuclear_ws, orbital_ws, filename):
 def extractOsaptData(filepath):
 
     vals = {}
-    vals['Elst']  = readBlock('%s/Elst.dat'  % filepath, H_to_kcal_)
-    vals['Exch']  = readBlock('%s/Exch.dat'  % filepath, H_to_kcal_)
-    vals['IndAB'] = readBlock('%s/IndAB.dat' % filepath, H_to_kcal_)
-    vals['IndBA'] = readBlock('%s/IndBA.dat' % filepath, H_to_kcal_)
-    vals['Disp']  = readBlock('%s/Disp.dat'  % filepath, H_to_kcal_)
+    vals['Elst']  = np.array(readBlock('%s/Elst.dat'  % filepath, H_to_kcal_))
+    vals['Exch']  = np.array(readBlock('%s/Exch.dat'  % filepath, H_to_kcal_))
+    vals['IndAB'] = np.array(readBlock('%s/IndAB.dat' % filepath, H_to_kcal_))
+    vals['IndBA'] = np.array(readBlock('%s/IndBA.dat' % filepath, H_to_kcal_))
+    # Read exact F-SAPT0 dispersion data
+    try:
+        vals['Disp'] = readBlock('%s/Disp.dat'  % filepath, H_to_kcal_) # Exact F-SAPT0 Dispersion
+    except FileNotFoundError:
+        print('No exact dispersion present.  Copying & zeroing `Elst.dat`->`Disp.dat`, and proceeding.\n')
```

Comment:
not `Edisp.dat -> Disp.dat`?

Reply:
I think copying from Elst.dat is the right way here, since Elst.dat and Disp.dat have the same dimension (involving orbitals), whereas EDisp.dat is always natom x natom. Whether we want to keep and print an all-zero Disp.dat is something we could avoid altogether I think.

Reply:
Ah, I understand now, you're just copying the shape. I was trying to avert a possible typo in the message. Thanks for the clarification.